### PR TITLE
improved test consistency for protected degree instance types

### PR DESCRIPTION
**Pull request by ketilaa**
Improved tests and readability regard protected degree/thesis publication instance types. It has been decided that Degree Licentiate also should be protected by the "thesis curator" access rights/permissions.

### CODE REVIEW

File: publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java [hunk 1]
```diff
@@ -20,14 +25,52 @@
...
+            Arguments.of(PublicationOperation.TERMINATE, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreeMaster.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreePhd.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeMaster.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreePhd.class)
+        );
+    }
```
Comment (truhacevkir) [inline publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java:56]: This one can be shortened, needs test adjustment.
```
    public static final List<Class<? extends DegreeBase>> DEGREES = List.of(DegreeLicentiate.class,
                                                                            DegreeBachelor.class,
                                                                            DegreeMaster.class,
                                                                            DegreePhd.class);

    private static Stream<Arguments> argumentsForDenyingCuratorFromPerformingOperationsOnProtectedDegreeResources() {
        return Stream.of(
            Arguments.of(PublicationOperation.UPDATE, DEGREES),
            Arguments.of(PublicationOperation.DELETE, DEGREES),
            Arguments.of(PublicationOperation.UNPUBLISH, DEGREES),
            Arguments.of(PublicationOperation.TERMINATE, DEGREES),
            Arguments.of(PublicationOperation.TICKET_PUBLISH, DEGREES)
        );
    }
```

**Review by truhacevkir — APPROVED** [review]
Following test is not stable:

`void shouldUpdateResourceWhenAuthorizedUserIsContributorAndHasCristinId()`

Failed when i was building with:

` User contributor@org was denied access UPDATE on publication 018ff2ef48f4-f99ec94f-1a01-4a3d-9eb7-a3fee7a48a00 from strategies NonDegreePermissionStrategy`

But it has nothing with this pr to do probably :)

File: publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java (hunk 1 above)
Comment (ketilaa) [reply publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java:56]: Then each iteration of the parameterised test would test multiple cases. I thank that is not something we want.

Comment (torbjokv) [reply publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java:56]: how about something like 
```java
    List<Class> degreeClasses = Arrays.asList(DegreeLicentiate.class, DegreeBachelor.class, DegreeMaster.class, DegreePhd.class);
    List<PublicationOperation> operations = Arrays.asList(PublicationOperation.UPDATE, PublicationOperation.DELETE, PublicationOperation.UNPUBLISH, PublicationOperation.TERMINATE, PublicationOperation.TICKET_PUBLISH);
    
    return operations.stream()
        .flatMap(operation -> degreeClasses.stream()
            .map(degreeClass -> Arguments.of(operation, degreeClass)))
        .collect(Collectors.toList())
        .stream();
```

File: publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java [hunk 2]
```diff
@@ -20,14 +25,52 @@
...
+            Arguments.of(PublicationOperation.UPDATE, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreeMaster.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreePhd.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeMaster.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreePhd.class)
+        );
+    }
```
Comment (torbjokv) [inline publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java:69]: same here

**Review by torbjokv — CHANGES_REQUESTED** [review]

**Review by torbjokv — APPROVED** [review]

**Pull request merged**